tasks/ChainPublisher.cpp

```diff
@@ -83,92 +29,15 @@ bool ChainPublisher::configureHook()
-    }
-
-    //Resize stuff
-    clear_and_resize_vectors();
-
-    //Receive Chains from urdf
-    for(size_t i=0; i<n_defined_chains_; i++){
-        std::string root = chain_definitions[i].root_link;
-
-        //By entering '__base__' as root or tip link, one could refer to the actual base-link of the robot model
-        if(root == "__base__"){
-            root = tree_.getRootSegment()->first;
-        }
-        std::string tip = chain_definitions[i].tip_link;
-        if(tip == "__base__"){
-            tip = tree_.getRootSegment()->first;
-        }
-
-        std::string name = chain_definitions[i].name;
-
-        st = tree_.getChain(root, tip, chains_[i]);
-        if(!st){
-            LOG_ERROR("Error extracting chain '%s' with root '%s' and tip '%s'. The urdf_file is: %s",
-                      name.c_str(), root.c_str(), tip.c_str(), urdf_file_path.c_str());
-            return false;
-        }
-        chain_names_[i] = name;
-    }
-
-    //Determine invloved joits, prepare frames storage and solvers
-    for(size_t i=0; i<n_defined_chains_; i++){
-        KDL::Chain chain = chains_[i];
-        //Determine involved joints for each chain
-        for(uint s=0; s<chain.segments.size(); s++){
-            KDL::Segment segment = chain.segments[s];
-            std::string jname = segment.getJoint().getName();
-
-            if(segment.getJoint().getType() == KDL::Joint::None){
-                LOG_DEBUG("Skipping joint %s of chain %s. Joint is fixed.",
-                          jname.c_str(),
-                          chain_names_[i].c_str());
-                continue;
-            }
-
-            involved_active_joints_[i].push_back(jname);
-        }
-        //Resize joint arrays
-        joint_arrays_[i].resize(involved_active_joints_[i].size());
-        joint_arrays_[i].data.setZero();
-
-        //Prepare frames storage
-        base::samples::RigidBodyState rbs;
-        if(chain_definitions[i].tip_link_renamed == ""){
-            rbs.sourceFrame = chain_definitions[i].tip_link;
-        }
-        else{
-            rbs.sourceFrame = chain_definitions[i].tip_link_renamed;
-        }
-        if(chain_definitions[i].root_link_renamed == ""){
-            rbs.targetFrame = chain_definitions[i].root_link;
-        }
-        else{
-            rbs.targetFrame = chain_definitions[i].root_link_renamed;
-        }
-        rbs.invalidate();
-        bt_frames_[i] = rbs;
-
-        //Prepare solver
-        pos_solvers_[i] = new KDL::ChainFkSolverPos_recursive(chains_[i]);
-    }
+    ChainTransformationCalculator transform_calculator(chain_definitions, urdf_file_path);
+    // assign the created object to the chain_transformer pointer to access this object in update_hook 
+    chain_transformer = &transform_calculator;
 
     //Create dynamic output per chain
-    for(int i=0; i<n_defined_chains_; i++){
-        std::string port_name = chain_names_[i];
+    for(int i=0; i< chain_transformer->get_number_chains(); i++){
```

Comment:
Can't you just iterate `chain_transformer->get_chain_names()` directly?

`for(const std::string& port_name :  chain_transformer->get_chain_names()){ ... }`